process/slash/multipleHeaderProposalsProof.go

```diff
@@ -0,0 +1,55 @@
+package slash
+
+import (
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/block/interceptedBlocks"
+)
+
+type multipleProposalProof struct {
+	level        SlashingLevel
+	slashingType SlashingType
+	headers      []*interceptedBlocks.InterceptedHeader
+}
+
+// NewMultipleProposalProof - creates a new double block proposal slashing proof with a level, type and data
+func NewMultipleProposalProof(sType SlashingType, level SlashingLevel, data []process.InterceptedData) (MultipleProposalProofHandler, error) {
+	headers, err := convertInterceptedDataToHeader(data)
+	if err != nil {
+		return nil, err
+	}
+
+	return &multipleProposalProof{
+		level:        level,
+		slashingType: sType,
+		headers:      headers,
+	}, nil
+}
+
+// GetLevel - gets the slashing proofs level
+func (mpp *multipleProposalProof) GetLevel() SlashingLevel {
+	return mpp.level
+}
+
+// GetType - gets the slashing proofs type
+func (mpp *multipleProposalProof) GetType() SlashingType {
+	return mpp.slashingType
+}
+
+// GetHeaders - gets the slashing proofs headers
+func (mpp *multipleProposalProof) GetHeaders() []*interceptedBlocks.InterceptedHeader {
+	return mpp.headers
+}
+
+func convertInterceptedDataToHeader(data []process.InterceptedData) ([]*interceptedBlocks.InterceptedHeader, error) {
```

Comment:
maybe rename it to convertInterceptedDataTo**Intercepted**Header

Reply:
Yes, it's better this way. Renamed it